components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt

```diff
@@ -37,17 +39,117 @@ class GeckoEngine(
 
     override fun name(): String = "Gecko"
 
+    /**
+     * See [Engine.errorTypeFromCode]
+     */
+    override fun errorTypeFromCode(errorCode: Int, errorCategory: Int?): ErrorPages.ErrorType {
```

Comment:
Maybe we can move this into a helper function or function extension for `ErrorType`? It'll reduce the complexity of this class (fixing that lint error in the CI build) and also makes it easier to write tests which having to mock the entire `GeckoEngine`.

Something like: `fun ErrorType.fromErrorInt(errorCode, errorCategory)`. 

I'm bad with function naming so feel free to come up with something better!